### ISSUE TITLE
KvsSample silently ignores label_smoothing_rate

### ISSUE THREAD

**Issue by LckyLke** [dice-group/dice-embeddings#437]
With smoothing 0.1, sampled labels should be [0.9, 0.1], consistent with the other sampled datasets. KvsSample instead returns [1.0, 0.0].

Reproduced on `66a7e619`.

```python
import numpy as np
import torch
from dicee.dataset_classes import KvsSampleDataset

ds = KvsSampleDataset(np.array([[0, 0, 1]]), {'a': 0, 'b': 1}, {'r': 0},
                      'EntityPrediction', neg_ratio=1, label_smoothing_rate=0.1)
labels = ds[0][2]
print(labels.tolist())  # Expected: [0.9, 0.1]
torch.testing.assert_close(labels, torch.tensor([0.9, 0.1]))
```
